optimize/backend/src/test/java/io/camunda/optimize/service/ProcessOverviewServiceTest.java

```diff
@@ -0,0 +1,194 @@
+    final String processKey = "processKey1";
+    final String processName = "Process 1";
+
+    final List<String> tenants = new ArrayList<>();
+    tenants.add("1");
+
+    final DefinitionWithTenantIdsDto definitionDto =
+        new DefinitionWithTenantIdsDto(processKey, processName, PROCESS, tenants, Set.of());
+    when(definitionService.getAllDefinitionsWithTenants(PROCESS))
+        .thenReturn(List.of(definitionDto));
+    when(definitionAuthorizationService.isAuthorizedToAccessDefinition(
+            userId, PROCESS, processKey, definitionDto.getTenantIds()))
+        .thenReturn(true);
+
+    final Map<String, ProcessOverviewDto> overviewMap =
+        Map.of(
+            processKey,
+            new ProcessOverviewDto(
+                "owner123", processKey, new ProcessDigestDto(false, Map.of()), Map.of()));
+    when(processOverviewReader.getProcessOverviewsByKey(Set.of(processKey)))
+        .thenReturn(overviewMap);
+
+    when(identityService.getIdentityNameById("owner123")).thenReturn(Optional.of("Owner Name"));
+
+    // when
+    final List<ProcessOverviewResponseDto> result =
+        processOverviewService.getAllProcessOverviews(userId, locale);
+
+    // then
+    assertThat(result).hasSize(1);
+    final ProcessOverviewResponseDto response = result.get(0);
+    assertThat(response.getProcessDefinitionKey()).isEqualTo(processKey);
+    assertThat(response.getProcessDefinitionName()).isEqualTo(processName);
+    assertThat(response.getOwner().getId()).isEqualTo("owner123");
+    assertThat(response.getOwner().getName()).isEqualTo("Owner Name");
+  }
+
+  @Test
+  void shouldUpdateProcessWhenAuthorizedAndValid() {
+    // given
+    final String userId = "user123";
+    final String processKey = "processKey1";
+    final String newOwnerId = "owner456";
+
+    final ProcessUpdateDto updateDto =
+        new ProcessUpdateDto(newOwnerId, new ProcessDigestRequestDto(true));
+
+    final List<String> tenants = new ArrayList<>();
+    tenants.add("1");
+
+    when(definitionService.getProcessDefinitionWithTenants(processKey))
+        .thenReturn(
+            Optional.of(
+                new DefinitionWithTenantIdsDto(
+                    processKey, "Process Name", PROCESS, tenants, Set.of())));
+    when(definitionAuthorizationService.isAuthorizedToAccessDefinition(any(), any(), any(), any()))
+        .thenReturn(true);
+
+    when(identityService.getUserById(newOwnerId)).thenReturn(Optional.of(new UserDto(newOwnerId)));
+    when(identityService.isUserAuthorizedToAccessIdentity(
+            userId, new IdentityDto(newOwnerId, IdentityType.USER)))
+        .thenReturn(true);
+
+    // when
+    processOverviewService.updateProcess(userId, processKey, updateDto);
+
+    // then
+    verify(processOverviewWriter).updateProcessConfiguration(processKey, updateDto);
+    verify(digestService).handleProcessUpdate(processKey, updateDto);
+  }
+
+  @Test
+  void shouldThrowExceptionWhenUnauthorizedForProcessUpdate() {
```

Comment:
missing tests:

- The happy path (no errors)
- User authorization fails (i.e. the stuff in `getValidatedOwnerId`)
- No owner set when digest is enabled (i.e. this is true: `(processUpdateDto.getProcessDigest().isEnabled() && processUpdateDto.getOwnerId() == null)`)

Reply:
fixed